app/models/application_record.rb

```diff
@@ -87,6 +87,22 @@ def self.aggregate_of(association)
     end
   end
 
+  def self.with_temporary_token(field_name, duration = 2.hours)
+    this = self
+    class_eval do
+      token_attribute = field_name
+      token_date_attribute = "#{field_name}_expiration_date"
+
+      define_method("generate_#{field_name}!") do
+        update!(token_attribute => this.generate_secure_token, token_date_attribute => duration.from_now)
+      end
+
+      define_method("#{field_name}_matches?") do |token|
+        token == attribute(token_attribute) && attribute(token_date_attribute) >= Time.now
```

Comment:
We should validate that the token is present to avoid possible security issues

Reply:
I can also add some kind of validation when this mixin is included. Any thoughts of which error should this throw?

Reply:
Do you mean something like throwing `NoMethodError` if those attributes are not defined?
I don't see it as a priority. It would happen anyway on invocation and I don't see a real benefit on doing this eagerly